server/src/main/java/org/elasticsearch/cluster/coordination/JoinHelper.java

```diff
@@ -83,22 +83,21 @@
     public static final String VALIDATE_JOIN_ACTION_NAME = "internal:cluster/coordination/join/validate";
     public static final String START_JOIN_ACTION_NAME = "internal:cluster/coordination/start_join";
 
-    // the timeout for each join attempt
```

Comment:
We probably don't need this for Zen1 joining too, although there's some complexity with the process of bootstrapping the cluster during a rolling upgrade. Rather than spend any time on the analysis I decided to leave the behaviour in mixed-version clusters alone.